docs/querying/api.md

```diff
@@ -1006,7 +1006,7 @@ $ curl -X POST \
 *New in v2.1 and supports PUT from v2.9*
 
 ### Clean Tombstones
-CleanTombstones removes the deleted data from disk and cleans up the existing tombstones. This can be used after deleting series to free up space.
+CleanTombstones removes the deleted data from disk and cleans up the existing tombstones. This can be used after deleting series to free up space. This process involves rerwiting existing blocks that have tombstones which requires free disk space (upto double the used space if all blocks have tombstones).
```

Comment:
```suggestion
CleanTombstones removes the deleted data from disk and cleans up the existing tombstones. This can be used after deleting series to free up space. This process involves rewriting existing blocks that have tombstones (aka normal compaction). Note that like normal compaction, this requires some extra disk space (one more TSDB block maximum).
```

Reply:
I don't think this belongs here at all, it's an odd place to put general TSDB workings. We already have the Storage page for operational considerations.

Reply:
@brian-brazil My experience purely as an operator, not a contributor and not someone with an in depth knowledge of TSDB was.

1. Performance was sluggish & we were low on disk space
1a. Turns out a bug in *our* app was had been generating a metric with very high cardinality in its labels
2. How do I delete these metrics; I get to https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series
3. That then leads me to this section about CleanTombstones
4. We run out of disk space

While sure the details of TSDB are should be doc, It would have helped my experience and probably that of others if there's at least some warning here that more disk space will be used, maybe even a link to the correct section of the TSDB page.

Reply:
Cleaning tombstones doesn't use any extra disk compared to normal operations, you'd have ended up using that much disk space anyway on the next big compaction. So while unfortunate, that you ran out of resources wasn't due to using CleanTombstones but rather the cardinality issue.